src/main/java/emu/grasscutter/data/excels/GadgetInteractData.java

```diff
@@ -0,0 +1,42 @@
+package emu.grasscutter.data.excels;
+
+import emu.grasscutter.data.GameResource;
+import emu.grasscutter.data.ResourceType;
+import lombok.Data;
+import lombok.Getter;
+
+@ResourceType(name = "GadgetInteractExcelConfigData.json")
+@Getter
+public class GadgetInteractData extends GameResource {
+    private int interactId;
+    private String actionType;
```

Comment:
for this and GadgetInteractAction  we could probably move it to an enum like this
```java
public enum InteractActionType {
    INTERACT_ACTION_NONE,
    INTERACT_ACTION_STATE,
    INTERACT_ACTION_SET_GADGET_CHAIN_BUFF,
    INTERACT_ACTION_UNLOCK_SPECIAL_TRANSPORT_POINT,
    INTERACT_ACTION_CONSUME_REGIONAL_PLAY_VAR
} 

```

This class also seems to be missing those fields:
```
private bool isGuestInteract;
private bool isMpModeInteract;
```

Reply:
First part: Added.

Second part: oh! odd!
In my old resources, isGuestInteract was called consumeItemId and isMpModeInteract was called consumeItemNum.
I'll make that change.